contracts/libraries/ERC721Booking.sol

```diff
@@ -30,6 +30,60 @@ abstract contract ERC721Booking is Context, ERC165, IERC721, IERC721Metadata, Re
         return bytes(baseURI).length > 0 ? string(abi.encodePacked(baseURI, tokenId.toString())) : "";
     }
 
+    /*============================================================
+                        BOOKING DATA STORAGE
+    ============================================================*/
+    struct Booking {
+        uint256 checkIn;
+        uint256 checkOut;
+        bytes data;
+    }
+
+    uint256 public bookingCounter;
+
+    mapping(uint256 tokenId => uint256) public bookingIds;
+
+    mapping(uint256 bookingId => Booking) public bookings;
+
+    function _createBooking(uint256 fromId, uint256 toId, bytes memory data) internal virtual {
+        unchecked {
+            bookingCounter++;
+        }
+
+        bookings[bookingCounter] = Booking(fromId, toId, data);
+        if (fromId == toId) {
```

Comment:
I think you don't need an `if () {} else { while () {} }` here. A `while () {}` would suffice.  Are you doing things in this way to avoid creating and assigning the `tokenId` local variable and save some gas? If so, I think we don't need to optimize so much. I would prefer code simplicity in this case, unless the gas difference is very high.